src/ui/components/ExperimentBrowserPage.tsx

```diff
@@ -0,0 +1,114 @@
+import { FC, useCallback, useEffect, useState } from "react";
+import { NimbusExperiment } from "@mozilla/nimbus-schemas";
+
+const PROD_URL =
+  "https://experimenter.services.mozilla.com/api/v6/experiments/";
+const STAGE_URL =
+  "https://stage.experimenter.nonprod.dataops.mozgcp.net/api/v6/experiments/";
+
+type Status = "Live" | "Preview";
+
+enum Environment {
+  Prod = "prod",
+  Stage = "stage",
+}
+
+const ExperimentBrowserPage: FC = () => {
+  const [environment, setEnvironment] = useState<Environment>(Environment.Prod);
+  const [status, setStatus] = useState<Status>("Live");
+  const [experiments, setExperiments] = useState<NimbusExperiment[]>([]);
+
+  const fetchExperiments = useCallback(
+    async (forceRefresh = false) => {
+      let url = environment === Environment.Prod ? PROD_URL : STAGE_URL;
+      url += `?status=${status}`;
+      if (forceRefresh) {
+        url += `&bust-cache=${Date.now()}`;
+      }
+
+      try {
+        const fetchedExperiments = await fetch(url).then(
+          (rsp) => rsp.json() as Promise<NimbusExperiment[]>,
+        );
+        setExperiments(fetchedExperiments);
+      } catch (error) {
+        console.error("Error fetching experiments:", error);
+      }
+    },
+    [environment, status],
+  );
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await fetchExperiments();
+    };
+    void fetchData();
+  }, [environment, status, fetchExperiments]);
```

Comment:
`fetchExperiments` will change if `environment` or `status` change, so there's no need to include them here.